### PR TITLE
Refactor + fix i -> j

### PR DESCRIPTION
**Pull request by 304NotModified**
@snakefoot 

I've refactored a bit, let me know if you like it

### CODE REVIEW

File: src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs [hunk 1]
```diff
@@ -33,9 +33,9 @@ public NLogMessageParameterList(IReadOnlyList<KeyValuePair<string, object>> para
         }
 
         /// <summary>
-        /// Attempts to parse the input parameterList
+        /// Create a <see cref="NLogMessageParameterList"/> if <paramref name="parameterList"/> has values, otherwise <c>null</c>
         /// </summary>
-        public static NLogMessageParameterList TryParseList(IReadOnlyList<KeyValuePair<string, object>> parameterList)
+        public static NLogMessageParameterList CreateIfNeeded(IReadOnlyList<KeyValuePair<string, object>> parameterList)
```
Comment (304NotModified) [inline src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs]: note: it won't parse IMO, so renamed it to create. Could be also "LazyCreate"

Comment (snakefoot) [reply src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs]: Well the NLogMessageParameterList-constructor initiates all the parsing/scanning. And the static-method calls the constructor, so I think TryParse is good.

Comment (snakefoot) [reply src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs]: If you rename back to TryParseList, then I will merge.

Comment (304NotModified) [reply src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs]: Done, added a `<remarks>` for your remark

Comment (304NotModified) [reply src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs]: oops, not proper pushed :(

Comment (304NotModified) [reply src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs]: https://github.com/snakefoot/NLog.Extensions.Logging/pull/2

**Comment by snakefoot** [comment]
Not needed already fixed. This PR will just break build


Sent from my Samsung device


-------- Original message --------

**Pull request merged**